### ISSUE TITLE
null is considered invalid for unspecified keys

### ISSUE THREAD

**Issue by shvarnty** [jquense/yup#895]
Hi and thank you for providing an excellent library...

when validating an object with extra fields unknown to the schema, those extra fields are considered invalid when null, but valid when undefined.

Example:
const schema = object().shape({
  someKnownFIeld: string.required()
})

shema.isValid({someKnownFIeld:"valid value", someUnkownField:undefined}) //**valid**
shema.isValid({someKnownFIeld:"valid value", someUnkownField:null}) //**invalid**

I would expect that both of these are valid, at least they are in my use case. Is this a bug or is there currently a way to handle this? I have tried many different things and nothing seems to work, unless I modify the object before passing, and setting all nulls to undefined.

I am using version 0.28.3

Thank you!

**Comment by jquense** [comment]
Yes `null` is not considers to unspecified value, add `nullable()` to your string schema if you want to accept null

**Comment by shvarnty** [comment]
If my Object has 100 fields, and I only want to validate 5 of those fields, I have to add the other 95 fields and say they are nullable? Can't I ignore fields that are not in my schema?

**Comment by jquense** [comment]
hmm i did notice it wasn't defined in the schema sorry. please use the issue template so i have a clearer picture about what is wrong

**Comment by dreamsinspace** [comment]
@shvarnty Shouldn't `object.noUnknown(false)` work?

**Comment by shvarnty** [comment]
After further testing I found that my issue was more nuanced, and the test I shared initially passes. 
TLDR, this was my issue, apologies.

My problem in case it helps others in the future:
My issue was that I missed an "otherwise" call inside of a "when"

...
name: string().when('nameRequired', {
   is: 'yes',
  then: string().required()
)
...

Above failed for me when name was null, because I did not have "otherwise: string().nullable()"
... so it would pass for undefined but not null values.

Apologies, this was my mistake. Thanks!
